src/Storages/StorageMaterializedView.cpp

```diff
@@ -444,9 +444,6 @@ SinkToStoragePtr StorageMaterializedView::write(const ASTPtr & query, const Stor
 void StorageMaterializedView::drop()
 {
     auto table_id = getStorageID();
-    const auto & select_query = getInMemoryMetadataPtr()->getSelectQuery();
-    if (!select_query.select_table_id.empty())
-        DatabaseCatalog::instance().removeViewDependency(select_query.select_table_id, table_id);
 
     if (getInMemoryMetadataPtr()->sql_security_type == SQLSecurityType::DEFINER)
         ViewDefinerDependencies::instance().removeViewDependencies(table_id);
```

Comment:
Don't we need to move this to shutdown also?

Reply:
No, AFAIU we use `ViewDefinerDependencies` for a kind of integrity check to make sure that we don't have dependent views when we drop a definer user. So looks like we don't want to remove it on DETACH TABLE, only on DROP. But looks like a similar issue is possible. @pufit what do you think?